src/components/Syntaxes.jsx

```diff
@@ -2,19 +2,28 @@ import React from 'react';
 import { Scrollbars } from 'react-custom-scrollbars';
 import { connect } from 'react-redux';
 
-import Input from './common/Input';
+import regExpEscape from './../helpers';
 import * as actions from '../actions';
 
-class Syntaxes extends React.Component {
+class Syntaxes extends React.PureComponent {
   constructor(props) {
     super(props);
-    this.state = { activeIndex: -1 };
+    this.state = {
+      activeIndex: -1,
+      currentlyDisplayed: this.props.syntaxes || [],
+      isEmpty: false,
+    };
     this.onClickHandler = this.onClickHandler.bind(this);
+    this.filterSyntaxes = this.filterSyntaxes.bind(this);
   }
 
   componentDidMount() {
     const { dispatch } = this.props;
-    dispatch(actions.fetchSyntaxes);
+    dispatch(actions.fetchSyntaxes).then(({ syntaxes }) => {
+      this.setState({
```

Comment:
Seems we don't need to wait till all syntaxes will be fetched since our state will be automatically change when this moment come.